src/type/types.cpp

```diff
@@ -549,6 +554,115 @@ std::ostream& operator<<(std::ostream& os, const StatementType& type) {
 }
 
 //===--------------------------------------------------------------------===//
+// QueryType - String Utilities
+//===--------------------------------------------------------------------===//
+std::string QueryTypeToString(QueryType query_type) {
+  switch(query_type) {
+    case QueryType::QUERY_BEGIN:return "BEGIN";
+    case QueryType::QUERY_COMMIT:return "COMMIT";
+    case QueryType::QUERY_ROLLBACK:return "ROLLBACK";
+    case QueryType::QUERY_CREATE_DB:return "CREATE DATABASE";
+    case QueryType::QUERY_CREATE_INDEX: return "CREATE INDEX";
+    case QueryType::QUERY_CREATE_TABLE: return "CREATE TABLE";
+    case QueryType::QUERY_CREATE_TRIGGER: return "CREATE TRIGGER";
+    case QueryType::QUERY_DROP:return "DROP";
+    case QueryType::QUERY_INSERT:return "INSERT";
+    case QueryType::QUERY_SET: return "SET";
+    case QueryType::QUERY_SHOW: return "SHOW";
+    case QueryType::QUERY_UPDATE: return "UPDATE";
+    case QueryType::QUERY_ALTER: return "ALTER";
+    case QueryType::QUERY_DELETE: return "DELETE";
+    case QueryType::QUERY_COPY: return "COPY";
+    case QueryType::QUERY_ANALYZE: return "ANALYZE";
+    case QueryType::QUERY_RENAME: return "RENAME";
+    case QueryType::QUERY_PREPARE: return "PREPARE";
+    case QueryType::QUERY_EXECUTE: return "EXECUTE";
+    case QueryType::QUERY_SELECT: return "SELECT";
+    case QueryType::QUERY_OTHER: default: return "OTHER";
+  }
+}
+
+QueryType StringToQueryType(std::string str) {
+  std::unordered_map<std::string, QueryType> querytype_string_map {
+      {"BEGIN", QueryType::QUERY_BEGIN}, {"COMMIT", QueryType::QUERY_COMMIT},
+      {"ROLLBACK", QueryType::QUERY_ROLLBACK}, {"CREATE DATABASE", QueryType::QUERY_CREATE_DB},
+      {"CREATE INDEX", QueryType::QUERY_CREATE_INDEX}, {"CREATE TABLE", QueryType::QUERY_CREATE_TABLE},
+      {"DROP", QueryType::QUERY_DROP}, {"INSERT", QueryType::QUERY_INSERT},
+      {"SET", QueryType::QUERY_SET}, {"SHOW", QueryType::QUERY_SHOW},
+      {"SHOW", QueryType::QUERY_SHOW}, {"UPDATE", QueryType::QUERY_UPDATE},
+      {"ALTER", QueryType::QUERY_ALTER}, {"DELETE", QueryType::QUERY_DELETE},
+      {"COPY", QueryType::QUERY_COPY}, {"ANALYZE", QueryType::QUERY_ANALYZE},
+      {"RENAME", QueryType::QUERY_RENAME}, {"PREPARE", QueryType::QUERY_PREPARE},
+      {"EXECUTE", QueryType::QUERY_EXECUTE}, {"SELECT", QueryType::QUERY_SELECT},
+      {"CREATE TRIGGER", QueryType::QUERY_CREATE_TRIGGER},
+      {"OTHER", QueryType::QUERY_OTHER},
+  };
+  std::unordered_map<std::string, QueryType>::iterator it  = querytype_string_map.find(str);
+  if (it != querytype_string_map.end()) {
+    return it -> second;
+  } else {
+    return QueryType::QUERY_INVALID;
+  }
+}
+
+QueryType StatementTypeToQueryType(StatementType stmt_type, parser::SQLStatement* sql_stmt) {
```

Comment:
Seems like we will not modify sql_stmt in this function, please make it a `const parser::SQLStatement*`, also the hash map seems to be created each time we call this function.

Reply:
Fixed